rust/connlib/tunnel/src/sockets.rs

```diff
@@ -64,21 +65,21 @@ impl Sockets {
         Poll::Ready(Ok(()))
     }
 
-    pub fn try_send(&mut self, transmit: quinn_udp::Transmit) -> io::Result<()> {
-        match transmit.destination {
+    pub fn try_send(&mut self, transmit: snownet::Transmit) -> io::Result<()> {
+        match transmit.dst {
             SocketAddr::V4(dst) => {
                 let socket = self.socket_v4.as_mut().ok_or(io::Error::new(
                     io::ErrorKind::NotConnected,
                     format!("failed send packet to {dst}: no IPv4 socket"),
                 ))?;
-                socket.send(transmit);
+                socket.send(transmit)?;
             }
             SocketAddr::V6(dst) => {
                 let socket = self.socket_v6.as_mut().ok_or(io::Error::new(
                     io::ErrorKind::NotConnected,
                     format!("failed send packet to {dst}: no IPv6 socket"),
                 ))?;
-                socket.send(transmit);
+                socket.send(transmit)?;
             }
         }
```

Comment:
GitHub won't let me suggest it but could this work?

```rust
let socket = match transmit.dst {
    ...
}?;
socket.send(transmit)?;
```

Since they're both `Option<Socket>`